File: sample/src/main/java/com/lazygeniouz/house/ads/sample/fragments/DialogAd.java
package com.lazygeniouz.house.ads.sample.fragments;

import android.content.Context;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.appcompat.widget.SwitchCompat;

import com.google.android.material.button.MaterialButton;
import com.lazygeniouz.house.ads.HouseAdsDialog;
import com.lazygeniouz.house.ads.listener.AdListener;
import com.lazygeniouz.house.ads.sample.R;

public class DialogAd extends BaseFragment implements AdListener {
    public DialogAd() {
    }

    private SwitchCompat hideIfAppInstalled, usePalette, hideHeader, allCaps;
    private EditText cardCorner, ctaCorner;
    private HouseAdsDialog dialog;
    private TextView loading;

    @SuppressWarnings("ConstantConditions")
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.dialog, container, false);
        boolean isShowLocalAssets = getContext().getSharedPreferences("localAssets", Context.MODE_PRIVATE).getBoolean("value", false);

        if (isShowLocalAssets) dialog = new HouseAdsDialog(getContext(), R.raw.ad_assets);
        else dialog = new HouseAdsDialog(requireContext(), "https://lz-houseads.firebaseapp.com/houseAds/ads.json");
        dialog.setAdListener(DialogAd.this);

        SwitchCompat localAssets = rootView.findViewById(R.id.useLocalResources);

        hideIfAppInstalled = rootView.findViewById(R.id.hideIfInstalled);
        usePalette = rootView.findViewById(R.id.usePalette);
        hideHeader = rootView.findViewById(R.id.showHeader);
        allCaps = rootView.findViewById(R.id.isAllCaps);
        cardCorner = rootView.findViewById(R.id.cardCorner);
        ctaCorner = rootView.findViewById(R.id.ctaCorner);
        loading = rootView.findViewById(R.id.loading);


        localAssets.setChecked(getContext().getSharedPreferences("localAssets", Context.MODE_PRIVATE).getBoolean("value", false));
        localAssets.setOnCheckedChangeListener((buttonView, isChecked) -> {
            getContext().getSharedPreferences("localAssets", Context.MODE_PRIVATE).edit().putBoolean("value", isChecked).apply();
            getActivity().recreate();
        });

        MaterialButton loadAds = rootView.findViewById(R.id.load);
        loadAds.setOnClickListener(v -> {
            loading.setVisibility(View.VISIBLE);
            dialog.hideIfAppInstalled(hideIfAppInstalled.isChecked())
                    .usePalette(usePalette.isChecked())
                    .showHeaderIfAvailable(hideHeader.isChecked())
                    .setCardCorners(Integer.parseInt(cardCorner.getText().toString()))
                    .setCtaCorner(Integer.parseInt(ctaCorner.getText().toString()))
                    .ctaAllCaps(allCaps.isChecked())
                    .loadAds();
        });

        return rootView;
    }

    @Override
    public void onAdLoadFailed(Exception e) {
        loading.setVisibility(View.GONE);
        Toast.makeText(getContext(), "Error: " + e.getMessage(), Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onAdLoaded() {
        dialog.showAd();
    }

    @Override
    public void onAdClosed() {
        Toast.makeText(getContext(), "Ad Closed", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onAdShown() {
        loading.setVisibility(View.GONE);
        Toast.makeText(getContext(), "Ad Shown", Toast.LENGTH_SHORT).show();
    }

    @Override
    public void onApplicationLeft() {
        Toast.makeText(getContext(), "Application Left", Toast.LENGTH_SHORT).show();
    }
}

File: sample/src/main/java/com/lazygeniouz/house/ads/sample/fragments/DialogAd.kt
package com.lazygeniouz.house.ads.sample.fragments

import android.content.Context
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.EditText
import android.widget.TextView
import androidx.appcompat.widget.SwitchCompat
import com.google.android.material.button.MaterialButton
import com.lazygeniouz.house.ads.HouseAdsDialog
import com.lazygeniouz.house.ads.listener.AdListener
import com.lazygeniouz.house.ads.sample.R
import com.lazygeniouz.house.ads.sample.fragments.base.BaseFragment

class DialogAd : BaseFragment(), AdListener {
    private lateinit var hideIfAppInstalled: SwitchCompat
    private lateinit var usePalette: SwitchCompat
    private lateinit var hideHeader: SwitchCompat
    private lateinit var allCaps: SwitchCompat
    private lateinit var cardCorner: EditText
    private lateinit var ctaCorner: EditText
    private lateinit var dialog: HouseAdsDialog
    private lateinit var loading: TextView

    override fun getLayoutId(): Int = R.layout.dialog

    override fun onViewCreated(rootView: View, savedInstanceState: Bundle?) {
        val isShowLocalAssets = requireContext().getSharedPreferences("localAssets", Context.MODE_PRIVATE).getBoolean("value", false)
        dialog = if (isShowLocalAssets) HouseAdsDialog(requireActivity(), R.raw.ad_assets) else HouseAdsDialog(requireActivity(), "https://lz-houseads.firebaseapp.com/houseAds/ads.json")
        dialog.setAdListener(this@DialogAd)

        val localAssets: SwitchCompat = rootView.findViewById(R.id.useLocalResources)
        hideIfAppInstalled = rootView.findViewById(R.id.hideIfInstalled)
        usePalette = rootView.findViewById(R.id.usePalette)
        hideHeader = rootView.findViewById(R.id.showHeader)
        allCaps = rootView.findViewById(R.id.isAllCaps)
        cardCorner = rootView.findViewById(R.id.cardCorner)
        ctaCorner = rootView.findViewById(R.id.ctaCorner)
        loading = rootView.findViewById(R.id.loading)
        localAssets.isChecked = requireContext().getSharedPreferences("localAssets", Context.MODE_PRIVATE).getBoolean("value", false)
        localAssets.setOnCheckedChangeListener { _, isChecked ->
            requireContext().getSharedPreferences("localAssets", Context.MODE_PRIVATE).edit().putBoolean("value", isChecked).apply()
            requireActivity().recreate()
        }
        val loadAds: MaterialButton = rootView.findViewById(R.id.load)
        loadAds.setOnClickListener {
            loading.visibility = View.VISIBLE
            dialog.apply {
                hideIfAppInstalled(hideIfAppInstalled.isChecked)
                usePalette(usePalette.isChecked)
                showHeaderIfAvailable(hideHeader.isChecked)
                setCardCorners(cardCorner.text.toString().toInt())
                setCtaCorner(ctaCorner.text.toString().toInt())
                ctaAllCaps(allCaps.isChecked)
                loadAds()
            }
        }
    }

    override fun onAdFailedToLoad(exception: Exception) {
        loading.visibility = View.GONE
        Log.d("HouseAdsExample", "Error: ${exception.message}")
    }

    override fun onAdLoaded() {
        dialog.showAd()
    }

    override fun onAdClosed() {
        Log.d("HouseAdsExample", "Ad Closed")
    }

    override fun onAdShown() {
        loading.visibility = View.GONE
        Log.d("HouseAdsExample", "Ad Shown")
    }

    override fun onApplicationLeft() {
        Log.d("HouseAdsExample", "Application Left")
    }
}